### PR TITLE
configuration and installer image changes

### PR DESCRIPTION
**Pull request by dorileo**
This PR proposes:
- replacing the mk-installer-image.sh script to a yaml descriptor (with its own post-install hook), though using installer to create the installer image - covering all the image creation aspects;
- support the use of both device file and image at the same time
- increased model's test coverage

### CODE REVIEW

**Comment by dorileo** [comment]
@mdhorn updated adding the --block-device argument. Also updated the Readme.md file.

**Comment by dorileo** [comment]
@mdhorn just found a couple of issues that will require my attention.

File: scripts/installer-image.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    mountpoint: /
+    size: "958M"
+    type: part
+
+bundles: [os-core, os-core-update, telemetrics, os-installer, openssh-server]
+telemetry: false
+keyboard: us
+language: en_US.UTF-8
+kernel: kernel-native
+
```
Comment (mdhorn) [inline scripts/installer-image.yaml]: This config should add:
```
postReboot: false
postArchive: false
autoUpdate: false
```
then the --reboot false not needed on the command line
We do not want to store the log file and the configuration file in the target installer image
We do not want the image to update to the lastest.

Comment (mdhorn) [reply scripts/installer-image.yaml]: @dorileo should we drop the '--reboot=false' from the Readme.md?

Comment (dorileo) [reply scripts/installer-image.yaml]: sure

Comment (dorileo) [reply scripts/installer-image.yaml]: done, thanks.

**Pull request merged**